sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -912,6 +912,14 @@ object SQLConf {
       .intConf
       .createWithDefault(10000)
 
+  val PANDAS_TIMEZONE_AWARE =
```

Comment:
There are other parts of the pyspark that doesn't use session local timezone. For instance, df.collect() and (maybe) python udf execution.

I am worried that having those to be inconsistent (some use local timezone, some doesn't) and complex (one configuration for each of these functionality?)

While it will be harder to fix, but how about we use one configuration to control the behavior of `df.toPandas()` and `df.collect()` and python udf regarding session local timezone?